std-bits/table/src/main/java/org/enso/table/util/NameDeduplicator.java

```diff
@@ -1,29 +1,73 @@
 package org.enso.table.util;
 
+import org.enso.table.problems.Problem;
+import org.enso.table.util.problems.DuplicateNames;
+import org.enso.table.util.problems.InvalidNames;
+
+import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class NameDeduplicator {
-  public static List<String> deduplicate(List<String> names) {
-    return deduplicate(names, " ");
+  private final Set<String> usedNames = new HashSet<>();
+  private final List<String> invalidNames = new ArrayList<>();
+  private final List<String> duplicatedNames = new ArrayList<>();
+
+  public String makeValid(String input) {
+    if (input == null || input.isEmpty()) {
+      this.invalidNames.add(input);
+      return "Column";
+    }
+
+    return input;
+  }
+
+  public List<String> makeUnique(List<String> names) {
+    return names.stream().map(this::makeUnique).collect(Collectors.toList());
+  }
+
+  public String makeUnique(String name) {
+    String validName = makeValid(name);
+    int currentIndex = validName.equals(name) ? 0 : 1;
```

Comment:
I'd add a comment on why this `0 : 1` switch happens.

I do understand it after reading the whole code and looking at the tests, but it's not immediately clear and I think it will be very helpful for anyone reading this code in the future if this is explained.

In general I think some short docstring explaining the desired behaviour could be helpful.